sensor/tests/helper/helper.go

```diff
@@ -431,6 +431,26 @@ func (c *TestContext) LastResourceStateWithTimeout(matchResourceFn MatchResource
 	}
 }
 
+// WaitForHello will wait until sensor transmits a `SensorHello` message to central and returns that message.
+func (c *TestContext) WaitForHello(timeout time.Duration) *central.SensorHello {
+	ticker := time.NewTicker(defaultTicker)
+	timeoutTimer := time.NewTicker(timeout)
+	for {
+		select {
+		case <-timeoutTimer.C:
+			c.t.Errorf("timeout (%s) reached waiting for hello event", timeout)
+			return nil
+		case <-ticker.C:
+			messages := c.GetFakeCentral().GetAllMessages()
+			for _, m := range messages {
+				if m.GetHello() != nil {
+					return m.GetHello()
+				}
+			}
+		}
+	}
```

Comment:
This construct looks a bit like `assert.Eventually` https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually. Have you given it a try in the past? Would it fit here maybe?

Reply:
I haven't tried that yet. Will check